graphql_compiler/tests/test_compiler.py

```diff
@@ -5423,6 +5423,74 @@ def test_fold_on_many_to_one_edge(self):
             expected_postgresql,
         )
 
+    def test_fold_after_recurse(self):
+        test_data = test_input_data.fold_after_recurse()
+
+        expected_postgresql = SKIP_TEST
+        expected_mssql = """
+        WITH anon_1(lives_in, parent, uuid, __cte_key, __cte_depth) AS (
+            SELECT
+                [Animal_1].lives_in AS lives_in,
+                [Animal_1].parent AS parent,
+                [Animal_1].uuid AS uuid,
+                [Animal_1].uuid AS __cte_key,
+                0 AS __cte_depth
+            FROM
+                db_1.schema_1.[Animal] AS [Animal_1]
+            UNION ALL
+            SELECT
+                [Animal_2].lives_in AS lives_in,
+                [Animal_2].parent AS parent,
+                [Animal_2].uuid AS uuid,
+                anon_1.__cte_key AS __cte_key,
+                anon_1.__cte_depth + 1 AS __cte_depth
+            FROM
+                anon_1 JOIN db_1.schema_1.[Animal] AS [Animal_2]
+                    ON anon_1.uuid = [Animal_2].parent
+            WHERE
+                anon_1.__cte_depth < 3
+        )
+        SELECT
+            [Animal_1].name AS animal_name,
+            folded_subquery_1.fold_output_name AS homes_list
+        FROM
+            db_1.schema_1.[Animal] AS [Animal_1],
```

Comment:
`[Animal_1]` should be joined to the rest, or `animal_name` should be picked from `anon_1`. See https://github.com/kensho-technologies/graphql-compiler/issues/866

Fix coming in separate PR.